### DISCUSSION TITLE
dashboard/dataset export raises schema issues

### DISCUSSION THREAD

**Discussion by LeiZhangImo** [apache/superset#18336] [Q&A / Help]
#### How to reproduce the bug

1. Go to swagger/v1 scroll down to dataset/export
2. Click on 'Try it out'
3. Put 1 in the 'q' filed

### Expected results

get 200 without error

### Actual results
When I use swagger for testing dataset export API in
mysupersite.com/swagger/v1
under the dataset/export part, after putting 1 in the 'q' filed, I'm getting this error:
**{
  "message": "Not a valid rison schema '1' is not of type 'integer'\n\nFailed validating 'type' in schema['items']:\n    {'type': 'integer'}\n\nOn instance[0]:\n    '1'"
}**

#### Screenshots
![image](https://user-images.githubusercontent.com/8062171/138454825-bfab76a1-4b75-46ce-95d5-b2b679510dbf.png)
![image](https://user-images.githubusercontent.com/8062171/138455091-b0981636-1b45-4758-951d-3d0a981042a5.png)

### Environment

- browser type and version: Chrome/94.0.4606.61
- superset version:  superset docker: apache/superset:1.3.2
- python version: Python 3.8.12
- node.js version: not know
- any feature flags active: connected to aws Anthena

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ x] I have reproduced the issue with at least the latest released version of superset.
- [x ] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

I'm using the Admin user

**Comment by StpMax** [comment]
Looks like this is issue in one of Superset dependencies. Here is request url that swagger compose:
```
/api/v1/dataset/export/?q=%5B%221%22%5D
```
`%5B%221%22%5D` is really `["1"]`. Value send as string, instead of integer. I tend to think that the issue is in `flask-appbuilder` or in swagger itself.

**Comment by LeiZhangImo** [comment]
> Looks like this is issue in one of Superset dependencies. Here is request url that swagger compose:
> 
> ```
> /api/v1/dataset/export/?q=%5B%221%22%5D
> ```
> 
> `%5B%221%22%5D` is really `["1"]`. Value send as string, instead of integer. I tend to think that the issue is in `flask-appbuilder` or in swagger itself.

Thank for your reply,
however, in the error log, it complains '1' is not type of 'integer instead of '["1"]' is not type of 'interger'

**Comment by jayakrishnankk** [comment]
Try this instead : `%5B1%2C%222%22%5D` , you will get error message about '2'. 1 will be successful since 1 is integer in this encoded string. Try the given string in https://meyerweb.com/eric/tools/dencoder/

**Comment by LeiZhangImo** [comment]
> Try this instead : `%5B1%2C%222%22%5D` , you will get error message about '2'. 1 will be successful since 1 is integer in this encoded string. Try the given string in https://meyerweb.com/eric/tools/dencoder/

yes, you are right, the API work if do url encode my myself, the there must be some swagger bug.

**Comment by LeiZhangImo** [comment]
after I manually fixed this error, the import/create API aways get ERROR:
https://github.com/apache/superset/issues/17206

**Comment by dpgaspar** [comment]
This endpoint is expecting `/api/v1/dataset/export/?q=(1,2,3)` for example, the REST API uses Rison parameters or a JSON encoded string. Looks like swagger is sending a JSON encoded string. I'll look into finding a solution

**Comment by jayakrishnankk** [comment]
> This endpoint is expecting `/api/v1/dataset/export/?q=(1,2,3)` for example, the REST API uses Rison parameters or a JSON encoded string. Looks like swagger is sending a JSON encoded string. I'll look into finding a solution

@dpgaspar - wondering if you found a solution to this.

**Reply by jorickdefraine** [reply]
I spent some time to find how to export a dataset with its id and about this endpoint: `/api/v1/dataset/export/?q=(1,2,3)` , I had to replace `(1,2,3)` with `[1,2,3]` to make it work.
Actually, It doesn't work if you use the swagger `Try it out` feature but you can send a get request with the endpoint above and it should work.
